controllers/idler/idler_controller_test.go

```diff
@@ -583,7 +583,195 @@ func TestEnsureIdlingFailed(t *testing.T) {
+						return pod
+					}
+				}
+			}
+			return nil
+		}()
+		//when
+		appType, appName, deletedByController, err := reconciler.scaleControllerToZero(logf.FromContext(context.TODO()), p.ObjectMeta)
+		//then
+		require.NoError(t, err)
+		require.Equal(t, true, deletedByController)
+		require.Equal(t, "StatefulSet", appType)
+		require.Equal(t, payloads.statefulSet.Name, appName)
+
+	})
+	t.Run("Test AppName/AppType for 'DeploymentConfig' ", func(t *testing.T) {
+		//given
+		p := func() *corev1.Pod {
+			for _, pod := range payloads.controlledPods {
+				for _, owner := range pod.OwnerReferences {
+					if owner.Kind == "ReplicationController" && owner.Name == fmt.Sprintf("%s-replicationcontroller", payloads.deploymentConfig.Name) {
+						return pod
+					}
+				}
+			}
+			return nil
+		}()
+		//when
+		appType, appName, deletedByController, err := reconciler.scaleControllerToZero(logf.FromContext(context.TODO()), p.ObjectMeta)
+		//then
+		require.NoError(t, err)
+		require.Equal(t, true, deletedByController)
+		require.Equal(t, "DeploymentConfig", appType)
+		require.Equal(t, payloads.deploymentConfig.Name, appName)
+
+	})
+	t.Run("Test AppName/AppType for 'ReplicationController' ", func(t *testing.T) {
+		//given
+		p := func() *corev1.Pod {
+			for _, pod := range payloads.allPods {
+				for _, owner := range pod.OwnerReferences {
+					if owner.Kind == "ReplicationController" && owner.Name == payloads.replicationController.Name {
+						return pod
+					}
+				}
+			}
+			return nil
+		}()
+		//when
+		appType, appName, deletedByController, err := reconciler.scaleControllerToZero(logf.FromContext(context.TODO()), p.ObjectMeta)
+		//then
+		require.NoError(t, err)
+		require.Equal(t, true, deletedByController)
+		require.Equal(t, "ReplicationController", appType)
+		require.Equal(t, payloads.replicationController.Name, appName)
+
+	})
+	t.Run("Test AppName/AppType for 'Job' ", func(t *testing.T) {
+		//given
+		p := func() *corev1.Pod {
+			for _, pod := range payloads.allPods {
+				for _, owner := range pod.OwnerReferences {
+					if owner.Kind == "Job" && owner.Name == payloads.job.Name {
+						return pod
+					}
+				}
+			}
+			return nil
+		}()
+		//when
+		appType, appName, deletedByController, err := reconciler.scaleControllerToZero(logf.FromContext(context.TODO()), p.ObjectMeta)
+		//then
+		require.NoError(t, err)
+		require.Equal(t, true, deletedByController)
+		require.Equal(t, "Job", appType)
+		require.Equal(t, payloads.job.Name, appName)
+
+	})
+	t.Run("Test AppName/AppType for 'Individual Pods' ", func(t *testing.T) {
+		//given
+		p := func() *corev1.Pod {
+			for _, pod := range payloads.standalonePods {
+				if pod.OwnerReferences == nil {
+					return pod
+				}
+			}
+			return nil
+		}()
+		//when
+		appType, appName, deletedByController, err := reconciler.scaleControllerToZero(logf.FromContext(context.TODO()), p.ObjectMeta)
+		if appName == "" {
+			appName = p.Name
+			appType = "Pod"
+		}
```

Comment:
We can't unfortunately test the individual pod case using `scaleControllerToZero()` function because it's not covered by that function. I'm afraid we need to test this case separately. You could prepare the reconcile by creating an individual Pod only (no other pods or controllers) and then reconcile it. Then check that the notification was created and that it's set to Pod kind and its name. It will also cover the functionality which actually sets the appName and appType in the notification object.